src/osd/OSDMapMapping.h

```diff
@@ -247,6 +247,15 @@ class OSDMapMapping {
 	row[4 + size + i] = up[i];
       }
     }
+
+    uint64_t get_num_acting_pgs() {
+      uint64_t num_acting_pgs = 0;
+      for (size_t ps = 0; ps < pg_num; ++ps) {
+        const int32_t *row = &table[row_size() * ps];
```

Comment:
could extract `row_size()` out, and cache it.

Reply:
fixed